cosmwasm/contracts/ibc-translator/src/reply.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    msg::GatewayIbcTokenBridgePayload,
+    state::{CHAIN_TO_CHANNEL_MAP, CURRENT_TRANSFER, CW_DENOMS},
+};
+use anybuf::Anybuf;
+use anyhow::{ensure, Context};
+use cosmwasm_std::{
+    from_binary, Binary, CosmosMsg::Stargate, Deps, DepsMut, Env, Reply, Response, SubMsg,
+};
+use cw_token_bridge::msg::CompleteTransferResponse;
+use wormhole_bindings::tokenfactory::{TokenFactoryMsg, TokenMsg};
+
+pub fn handle_complete_transfer_reply(
+    deps: DepsMut,
+    env: Env,
+    msg: Reply,
+) -> Result<Response<TokenFactoryMsg>, anyhow::Error> {
+    // we should only be replying on success
+    ensure!(
+        msg.result.is_ok(),
+        "msg result is not okay, we should never get here"
+    );
+
+    let res_data_raw = cw_utils::parse_reply_execute_data(msg)
+        .context("failed to parse protobuf reply response_data")?
+        .data
+        .context("no data in the response, we should never get here")?;
+    let res_data: CompleteTransferResponse =
+        from_binary(&res_data_raw).context("failed to deserialize response data")?;
+    let contract_addr = res_data
+        .contract
+        .context("no contract in response, we should never get here")?;
+
+    // load interim state
+    let transfer_info = CURRENT_TRANSFER
+        .load(deps.storage)
+        .context("failed to load current transfer from storage")?;
+
+    // delete interim state
+    CURRENT_TRANSFER.remove(deps.storage);
+
+    // deserialize payload into the type we expect
+    let payload: GatewayIbcTokenBridgePayload = serde_json_wasm::from_slice(&transfer_info.payload)
+        .context("failed to deserialize transfer payload")?;
+
+    match payload {
+        GatewayIbcTokenBridgePayload::Simple {
+            chain,
+            recipient,
+            fee: _,
```

Comment:
How is the relayer fee handled? It looks like in `convert_cw20_to_bank_and_send` that the recipient receives the full amount. Does the spec require that this fee be zero?

Reply:
For going from external chains to Cosmos chains, the `fee` parameter isn't used today. We primarily added the `fee` parameter for compatibility with the wormhole token bridge when going from Cosmos chains back out to external chains - when we call the token bridge, we need to provide a `fee`, which, if non-zero, is used to incentivize specialized relayers.